fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -186,6 +186,8 @@ where
     }
 
     async fn end(&self, mut state: Self::State) -> anyhow::Result<(Self::State, Self::EndOutput)> {
+        checkpoint::emit_trace_if_check_checkpoint_finalized(&self.gateway, &mut state)?;
```

Comment:
If tracing fails, we should degrade gracefully instead of propagating an error through the critical path. After all, tracing is actually dispensable. Options:

1. Skip the trace silently. However, this leads to inconsistency as users might have set up monitoring for it.
2. Log the error as a warning. This gives info to the user to go hunting once they figure out that the finalization is actually happening, and it's only the trace that's not being printed.
3. Add a meta-event as a fallback to capture tracing errors in the journal. This is my preferred solution, as it actively communicates there is an error and keeps the journal consistent.

If you agree, let's go with (3). I'd suggest adding the following event:

```
struct TracingError {
	affected_event: String,
	reason: String,
}
```

And using `.inspect_err()` here to emit that trace on error.

I'd also add Counter metric for this: `tracing_errors[event=$affected_event]`.

WDYT?

Reply:
Great point! I definitely think this should not propagate the error to the upper level. The best solution here, in my opinion, would be the proposed solution 3. The metrics are also a good idea, as they can help flag that the trace is not being saved without the need for proactive debugging.